salt/utils/parsers.py

```diff
@@ -304,7 +316,7 @@ def _mixin_setup(self):
         # the config options and if needed override them
         self._mixin_after_parsed_funcs.append(self.__merge_config_with_cli)
 
-    def __merge_config_with_cli(self, *args):  # pylint: disable=unused-argument
```

Comment:
why do we need to take `*args` here and everywhere else? Wouldn't they just be overridden in the subclass? 🤔

Reply:
We don't, hence it was removed.
Previously, because we were keeping bound methods on the call back lists, we were actually passing `self` twice.